python/dolfinx/__init__.py

```diff
@@ -6,28 +6,6 @@
 """Main module for DOLFINx"""
 
 # flake8: noqa
-
-# Store dl open flags to restore them after import
-import sys
-
-stored_dlopen_flags = sys.getdlopenflags()
-
-# Developer note: below is related to OpenMPI
-# Fix dlopen flags (may need reorganising)
-if "linux" in sys.platform:
```

Comment:
This need to be checked. My recollection os that the removed code was required when MKL is used.

Reply:
As it stands it won't work because `ctypes.RTLD_NOW` no longer exists, at least in any of my installs. There is `ctypes.RTLD_GLOBAL` and `ctypes.RTLD_LOCAL`.

The static checker picked this up as an error which was quite nice.

Reply:
@chrisrichardson do you know the purpose/necessity of this code path?